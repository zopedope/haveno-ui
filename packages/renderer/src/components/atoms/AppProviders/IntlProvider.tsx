// =============================================================================
//  Copyright 2022 Haveno
//
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
// =============================================================================

import { useEffect, useMemo, useState } from "react";
import { IntlProvider as ReacIntlProvider } from "react-intl";
import type { FC } from "react";
import { LangPack, SupportedLocales } from "@src/constants/lang";
import type { LangKeys } from "@src/constants/lang";

const DEFAULT_LOCALE = SupportedLocales.EN;

export const IntlProvider: FC = ({ children }) => {
  const [messages, setMessages] = useState<{ [key in LangKeys]: string }>();
  const locale = useMemo(
    () =>
      navigator.language in SupportedLocales
        ? navigator.language
        : DEFAULT_LOCALE,
    [navigator.language]
  );

  useEffect(() => {
    setMessages(LangPack[locale] ?? LangPack[DEFAULT_LOCALE]);
  }, [locale]);

  return (
    <ReacIntlProvider locale={locale} messages={messages}>
      {children}
    </ReacIntlProvider>
  );
};
