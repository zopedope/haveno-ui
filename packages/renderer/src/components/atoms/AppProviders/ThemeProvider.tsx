// =============================================================================
//  Copyright 2022 Haveno
//
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
// =============================================================================

import { MantineProvider, Global } from "@mantine/core";
import type { FC } from "react";
import { themeOverride, globalStyles } from "@src/theme";

export const ThemeProvider: FC = ({ children }) => {
  return (
    <MantineProvider theme={themeOverride}>
      <Global styles={globalStyles} />
      {children}
    </MantineProvider>
  );
};
