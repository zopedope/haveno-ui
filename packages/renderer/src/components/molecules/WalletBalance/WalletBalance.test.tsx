// =============================================================================
//  Copyright 2022 Haveno
//
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
// =============================================================================

import { beforeAll, describe, expect, it, vi } from "vitest";
import { render, waitForElementToBeRemoved } from "@testing-library/react";
import { WalletBalance } from ".";
import { AppProviders } from "@atoms/AppProviders";

describe("molecules::WalletBalance", () => {
  beforeAll(() => {
    vi.mock("@hooks/haveno/useBalances", () => ({
      useBalances: () => ({
        isLoading: false,
        isSuccess: true,
        data: {
          lockedBalance: 12,
          reservedTradeBalance: 14,
          balance: 15,
        },
      }),
    }));

    vi.mock("@hooks/storage/useAccountInfo", () => ({
      useAccountInfo: () => ({
        isLoading: false,
        isSuccess: true,
        data: {
          primaryFiat: "USD",
        },
      }),
    }));

    vi.mock("@hooks/haveno/usePrice", () => ({
      usePrice: () => ({
        isLoading: false,
        isSuccess: true,
        data: 300,
      }),
    }));
  });

  it("renders loading state", () => {
    const { asFragment, unmount } = render(
      <AppProviders>
        <WalletBalance />
      </AppProviders>
    );
    expect(asFragment()).toMatchSnapshot();
    unmount();
  });

  it("renders after loading data", async () => {
    const { asFragment, queryByText, unmount } = render(
      <AppProviders>
        <WalletBalance />
      </AppProviders>
    );
    if (queryByText("Loading...")) {
      await waitForElementToBeRemoved(() => queryByText("Loading..."));
    }
    expect(asFragment()).toMatchSnapshot();
    unmount();
  });
});
