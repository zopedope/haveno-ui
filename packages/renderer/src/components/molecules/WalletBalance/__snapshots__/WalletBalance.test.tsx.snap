// Vitest Snapshot v1

exports[`molecules::WalletBalance > renders after loading data 1`] = `
<DocumentFragment>
  <div
    class="mantine-Group-root mantine-cvr2kw"
  >
    <div
      class="mantine-Stack-root mantine-Group-child mantine-1vgxq13"
    >
      <div
        class="mantine-Text-root mantine-hbkn2v"
      >
        Available Balance
      </div>
      <div
        class="mantine-Text-root mantine-1swgem8"
      >
        0.00
      </div>
    </div>
    <div
      class="mantine-Stack-root mantine-Group-child mantine-1vgxq13"
    >
      <div
        class="mantine-Text-root mantine-hbkn2v"
      >
        Locked Funds
      </div>
      <div
        class="mantine-Text-root mantine-1swgem8"
      >
        12.00
      </div>
    </div>
    <div
      class="mantine-Stack-root mantine-Group-child mantine-1vgxq13"
    >
      <div
        class="mantine-Text-root mantine-hbkn2v"
      >
        Reserved Funds
      </div>
      <div
        class="mantine-Text-root mantine-1swgem8"
      >
        0.00
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`molecules::WalletBalance > renders loading state 1`] = `
<DocumentFragment>
  <div
    class="mantine-Group-root mantine-cvr2kw"
  >
    <div
      class="mantine-Stack-root mantine-Group-child mantine-1vgxq13"
    >
      <div
        class="mantine-Text-root mantine-hbkn2v"
      >
        Available Balance
      </div>
      <div
        class="mantine-Text-root mantine-1swgem8"
      >
        0.00
      </div>
    </div>
    <div
      class="mantine-Stack-root mantine-Group-child mantine-1vgxq13"
    >
      <div
        class="mantine-Text-root mantine-hbkn2v"
      >
        Locked Funds
      </div>
      <div
        class="mantine-Text-root mantine-1swgem8"
      >
        12.00
      </div>
    </div>
    <div
      class="mantine-Stack-root mantine-Group-child mantine-1vgxq13"
    >
      <div
        class="mantine-Text-root mantine-hbkn2v"
      >
        Reserved Funds
      </div>
      <div
        class="mantine-Text-root mantine-1swgem8"
      >
        0.00
      </div>
    </div>
  </div>
</DocumentFragment>
`;
